app/mailers/activity_mailer.rb

```diff
@@ -2,31 +2,19 @@
 #
 # @api private
 class ActivityMailer < ApplicationMailer
+  helper :application_notifications
+
   # Emails a recipient, informing him of an activity.
   #
   # @param [User] recipient The recipient of the email.
-  # @param object The object to be made available to the view, accessible using +@object+.
-  # @param [String] view_path The path to the view which should be rendered.
-  def email(recipient, object, view_path)
+  # @param [Course::Notification|UserNotification] notification The related notification
+  def email(recipient, notification)
     @recipient = recipient
-    @object = object
-    mail(to: recipient.email, template: view_path)
-  end
-
-  protected
-
-  # Adds support for the +template+ option, which specifies an absolute path.
-  #
-  # @option options [String] :template (nil) The absolute template path to render.
-  # @see #{ActionMailer::Base#mail}
-  def mail(options)
-    template = options.delete(:template)
-    if template
-      prepend_view_path(File.dirname(template))
-      options[:template_path] = ''
-      options[:template_name] = File.basename(template)
+    @object = notification.activity.object
+    view_path = view_context.notification_view_path(notification)
+    mail(to: recipient.email) do |format|
+      format.html { render view_path }
```

Comment:
I removed this for a reason. By doing this you remove the automatic template inference for HTML/plain text emails.